File: ServiceBus/MessagingExpress/Queue/all-ops.js
const { ServiceBusClient } = require("@azure/service-bus");

const connectionString = ""
const serviceBusClient = new ServiceBusClient(connectionString);
// create service bus administration client
const administrationClient = serviceBusClient.createAdministrationClient();

// create a queue
async function createQueue(queueName) {
    try {
        await administrationClient.createQueue(queueName);
    }
    catch(err) {
        console.log(err);
    }
}

// send message to the queue
async function sendMessage(queueName){
    try
    {
        const sender = serviceBusClient.createSender(queueName);
        
        const message1 = {
            body: "Hello, from Node to Queue.", 
            label: "greeting", 
            useProperties: { priority: 1}
        };

        const message2 = {
            body: "Hi, from Node to Queue.", 
            label: "greeting", 
            useProperties: { priority: 1}
        };

        //send an array of two messages to queue
        await sender.sendMessages([message1, message2]);
        await sender.close();
    }
    catch(err){
        console.log(err);
    }
}

// receive messages from a queue
async function receiveMessages(queueName){
    try
    {
        const receiver = serviceBusClient.createReceiver(queueName);
        // receive a batch of 2 messages and set peek lock of 3 seconds
        const receivedMessages = await receiver.receiveMessages(2, {maxWaitTimeInMs: 3000});

        // process received messages
        for (let i=0; i<receivedMessages.length; i++) {

            // do something with message
            console.log(receivedMessages[i]);

            // complete the message after processing
            await receiver.completeMessage(message);
        }

        await receiver.close();
    }
    catch(err){
        console.log(err);
    }
}

// update a queue
async function updateQueue(queueName) {
    try {
        // get current queue properties
        let queueProperties = await administrationClient.getQueue(queueName);
        // change max delivery count value
        queueProperties.maxDeliveryCount = 5;
        // update queue with modified property
        await administrationClient.updateQueue(queueProperties);
    }
    catch(err) {
        console.log(err);
    }
}

// delete a queue
async function deleteQueue(queueName) {
    try {
        await administrationClient.deleteQueue(queueName);
    }
    catch(err) {
        console.log(err);
    }
}

// main function to run program
async function main(queueName) {

    try {
        // create queue
        await createQueue(queueName);
        // send message
        await sendMessage(queueName);
        // receive messages
        await receiveMessages(queueName);
        // update queue
        await updateQueue(queueName);
        // delete queue
        await deleteQueue(queueName);
    }
    catch(err) {
        console.log(err);
    }
    finally {
        administrationClient.close();
        serviceBusClient.close();
    }
}

// call main function
await main("queue001");

File: ServiceBus/MessagingPython/Queue/all-ops.py
from azure.servicebus import ServiceBusClient, ServiceBusMessage
from azure.servicebus.management import ServiceBusAdministrationClient

connection_str = ""
queue_name = "queue001"

# create service bus client
servicebus_client = ServiceBusClient.from_connection_string(connection_str)

# create service bus administration client
servicebus_admin_client = ServiceBusAdministrationClient.from_connection_string(connection_str)

# create queue
servicebus_admin_client.create_queue(queue_name)

# update queue
queue_properties = servicebus_admin_client.get_queue(queue_name)
queue_properties.max_delivery_count = 10
servicebus_admin_client.update_queue(queue_properties)

# send a message to queue
sender = servicebus_client.get_queue_sender(queue_name)
message = ServiceBusMessage("Hello from Python")
sender.send_messages(message)
sender.close()
    
# receive message from queue
receiver = servicebus_client.get_queue_receiver(queue_name)
received_message = receiver.receive_messages(max_message_count=1)[0]
# do something with received message
print(received_message)
# mark message as completed
receiver.complete_message(received_message)
receiver.close()

# delete queue
servicebus_admin_client.delete_queue(queue_name)

# close admin client and service client
servicebus_admin_client.close()
servicebus_client.close()